aea/protocols/dialogue/base.py

```diff
@@ -338,6 +339,52 @@ def __init__(
         )
         self._message_class = message_class
 
+    def __eq__(self, other) -> bool:
+        """Compare two dialogues."""
+        return (
+            type(self) == type(other)  # pylint: disable=unidiomatic-typecheck
```

Comment:
why not use isinstance?

Reply:
the type has to be right the same, isinstsance(anydialogue, Basedialogue) == True, but that's not correct.